src/repl/scala/tools/nsc/interpreter/Imports.scala

```diff
@@ -127,7 +127,8 @@ trait Imports {
           case rh :: rest if !keepHandler(rh.handler) => select(rest, wanted)
           case rh :: rest                             =>
             import rh.handler._
-            val newWanted = wanted ++ referencedNames -- definedNames -- importedNames
+            val augment = rh match { case ReqAndHandler(_, _: ImportHandler) => referencedNames ; case _ => Nil }
```

Comment:
Yes, that's the new test, I think. I couldn't think of a reason to do it for arbitrary defs. I was hoping a broken test would tell me.